### ISSUE TITLE
`reportUninitializedInstanceVariable` false negative when assigning to non-existent attribute in a non-constructor method of a class that extends `ABC`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1210]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBAhgRgYygSwgBzCALlAggIQGEBYAKDIQBsYBnGqAMTDAAoDCBKALjKj6gAmAU2CwaA4CxpDKwbr36LpsgHS0J44FxQA7HAF4oARigBiKDrBQhIcCCA)

```python
from abc import ABC

class Foo(ABC):
    def asdf(self):
        self.asdfsdf: int = 1 # no error
```
